[Split CSSPropertyParserHelpers] Part 1: Primitives, Lists, and Color

weinig opened the pull request:
#### 6955e8112f272d0401cd1861da095015ba37f810
<pre>
[Split CSSPropertyParserHelpers] Part 1: Primitives, Lists, and Color
<a href="https://bugs.webkit.org/show_bug.cgi?id=272774">https://bugs.webkit.org/show_bug.cgi?id=272774</a>

Reviewed by Tim Nguyen.

CSSPropertyParserHelpers.cpp is enormous at 8872 lines long, which
makes it hard to work with and makes compile/fix cycles unwieldily.

This starts the process of breaking it up into logical chunks, starting
with the the primitive consumers that correspond to token types (number,
ident, percent, etc.), lists and color (color weighs in at ~1600 lines
all by itself).

No code changes are being made here, this is purely moving things around.

CSSPropertyParserHelpers.cpp goes from 8872 lines to 5776 lines, that&apos;s
3096 lines moved.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/ShorthandSerializer.cpp:
* Source/WebCore/css/parser/CSSCalcParser.cpp: Added.
* Source/WebCore/css/parser/CSSCalcParser.h: Added.
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Ident.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Ident.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+List.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Meta.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+None.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+None.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percent.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percent.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/query/GenericMediaQueryParser.cpp:
* Source/WebCore/page/IntersectionObserver.cpp:
* Source/WebCore/style/StyleBuilderState.h:

Canonical link: <a href="https://commits.webkit.org/277585@main">https://commits.webkit.org/277585@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/31a649c5fa0f6bb6410435a19723f8051d48ecf0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48038 "58 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27243 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50959 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50725 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44098 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50345 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33131 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24737 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48621 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24914 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20378 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22395 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/42728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6091 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/44384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43163 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52620 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23086 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/19438 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46575 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24351 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41704 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45445 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10600 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25151 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24073 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/page/IntersectionObserver.cpp:
```diff
@@ -25,11 +25,11 @@
 #include "IntersectionObserver.h"
 
 #include "CSSParserTokenRange.h"
 #include "CSSPropertyParserHelpers.h"
```
Do we still need this include?

weinig replied:
Nope. Removed.

nt1m commented on Source/WebCore/css/query/GenericMediaQueryParser.cpp:
```diff
@@ -29,6 +29,11 @@
+#include "CSSPropertyParserConsumer+Length.h"
+#include "CSSPropertyParserConsumer+Number.h"
+#include "CSSPropertyParserConsumer+Resolution.h"
 #include "CSSPropertyParserHelpers.h"
```
Do we still need to import `CSSPropertyParserHelpers` ?

weinig replied:
That is still needed.

nt1m commented on Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
```diff
@@ -40,6 +40,12 @@
+#include "CSSPropertyParserConsumer+Integer.h"
+#include "CSSPropertyParserConsumer+List.h"
+#include "CSSPropertyParserConsumer+Percent.h"
 #include "CSSPropertyParserHelpers.h"
```
Do we still need to import `CSSPropertyParserHelpers` ?

weinig replied:
That is still needed.

nt1m reviewed (approved):
Can you also address the style bot comments?

Commit:
[Split CSSPropertyParserHelpers] Part 1: Primitives, Lists, and Color
https://bugs.webkit.org/show_bug.cgi?id=272774

Reviewed by Tim Nguyen.

CSSPropertyParserHelpers.cpp is enormous at 8872 lines long, which
makes it hard to work with and makes compile/fix cycles unwieldily.

This starts the process of breaking it up into logical chunks, starting
with the the primitive consumers that correspond to token types (number,
ident, percent, etc.), lists and color (color weighs in at ~1600 lines
all by itself).

No code changes are being made here, this is purely moving things around.

CSSPropertyParserHelpers.cpp goes from 8872 lines to 5776 lines, that's
3096 lines moved.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/ShorthandSerializer.cpp:
* Source/WebCore/css/parser/CSSCalcParser.cpp: Added.
* Source/WebCore/css/parser/CSSCalcParser.h: Added.
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Ident.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Ident.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+List.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Meta.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+None.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+None.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percent.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percent.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/query/GenericMediaQueryParser.cpp:
* Source/WebCore/page/IntersectionObserver.cpp:
* Source/WebCore/style/StyleBuilderState.h:

Canonical link: https://commits.webkit.org/277585@main
